File: tourFinder/src/main/java/fmi/java/web/tourFinder/model/Tour.java
package fmi.java.web.tourFinder.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "tours")
public class Tour {
    @Id
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid")
    private String id;
    private String name;
    private String destination;
    private String description;
    private double price;
    private Date startDate;
    private Date endDate;
    private String AgencyName;
}

File: frontEnd/src/models/Tour.ts
export class Tour{
  public id: string;
  public name: string;
  public destination: string;
  public description: string;
  public price: number;
  public startDate: Date;
  public endDate: Date;
  public agencyName: string;

  constructor(id: string, name: string, destination: string, description: string, price: number, startDate: Date,
              endDate: Date, agencyName: string) {
    this.id = id;
    this.name = name;
    this.destination = destination;
    this.description = description;
    this.price = price;
    this.startDate = startDate;
    this.endDate = endDate;
    this.agencyName = agencyName;
  }
}
